sample/sample.go

```diff
@@ -0,0 +1,181 @@
+package sample
+
+import (
+	"errors"
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/stat/sampleuv"
+)
+
+type Sampler interface {
+	Sample([]float64) ([]float64, error)
+}
+
+type Temperature float64
+
+func (s Temperature) Sample(logits []float64) ([]float64, error) {
+	if s < 0 || s > 1 {
+		return nil, errors.New("temperature must be between 0 and 1")
+	}
+
+	copiedLogits := append([]float64(nil), logits...)
+	// Greedy sampling
+	if s == 0 {
+		return []float64{floats.Max(copiedLogits)}, nil
+	}
+	floats.Scale(1.0/float64(s), copiedLogits)
+	return copiedLogits, nil
+}
+
+type softmax struct{}
+
+func Softmax() Sampler {
```

Comment:
Why? Just export the type?

Reply:
The only thing after deleting this is that the usage looks a bit uglier and also breaks the consistency with the other methods.

Current:
```go
probs, err := Softmax().Sample([]float64{-3, -2, -1, 0, 1, 2, 4})
```

After removing:
```go
probs, err := softmax{}.Sample([]float64{-3, -2, -1, 0, 1, 2, 4})
```

For comparison - TopK usage:
```go
logits, err := TopK(3).Sample([]float64{-3, -2, -1, 0, 1, 2, 4})
```